gapir/cc/android/asset_resource_cache.cpp

```diff
@@ -48,13 +48,16 @@ bool asset_read(AAsset *asset, void *buf, size_t count) {
   return true;
 }
 
-// touch pages will touch all memory pages for a given memory span
+// touch_pages will touch all memory pages for a given memory span
 void touch_pages(void *addr, uint32_t size) {
-  long pagesize = sysconf(_SC_PAGESIZE);
+  static const long pagesize = sysconf(_SC_PAGESIZE);
   char *end = ((char *)addr) + size;
   for (char *p = (char *)addr; p < end; p += pagesize) {
     *p = '0';
   }
+  // Make sure the last page is touched, as the loop may exit without touching
+  // it when p lands in the last page to touch, but beyond end.
+  *(end - 1) = '0';
```

Comment:
if (size == 0) this will write out of bounds.

Reply:
Yes I was assuming this function to be called only for with non-zero size resource as parameter, but we'd better check this indeed.